### DISCUSSION TITLE
How to run new changes?

### DISCUSSION THREAD

**Discussion by TheAdventurer64** [organicmaps/organicmaps#3110] [Q&A]
In #3102, I added a new cuisine changes in a dedicated branch. By building the Organic Maps app, it would run using the code in the master branch, right? If that's the case, how would I test the changes in the dedicated branch? And would it be possible to test POI changes on the desktop app, or would I need to run a mobile emulator?

**Comment by TheAdventurer64** [comment]
@biodranik

**Reply by biodranik** [reply]
To test changes that modify our types system, you need to use generator_tool to build some map areas with your changes. And it's always better to rebase your changes on the most recent master branch, we constantly improve it and add new types too.
